pkg/cgroup/cgroup.go

```diff
@@ -14,73 +12,332 @@ import (
 	"github.com/aquasecurity/tracee/pkg/mount"
 )
 
-const CgroupV1Controller = "cpuset"
-const CgroupV1FsType = "cgroup"
-const CgroupV2FsType = "cgroup2"
+// Constants
 
-const CgroupCpusetReleaseAgent = "/sys/fs/cgroup/cpuset/release_agent"
+const (
+	CgroupV1FsType          = "cgroup"
+	CgroupV2FsType          = "cgroup2"
+	CgroupDefaultController = "cpuset"
+	CgroupReleaseAgent      = "/sys/fs/cgroup/cpuset/release_agent"
+	procCgroups             = "/proc/cgroups"
+)
+
+// Versions
+
+type CgroupVersion int
+
+func (v CgroupVersion) String() string {
+	switch v {
+	case CgroupVersion1:
+		return CgroupV1FsType
+	case CgroupVersion2:
+		return CgroupV2FsType
+	}
+
+	return ""
+}
+
+const (
+	CgroupVersionNone CgroupVersion = 1 << iota
+	CgroupVersion1
+	CgroupVersion2
+	CgroupVersionAll CgroupVersion = CgroupVersion1 & CgroupVersion2
+)
+
+//
+// Cgroups
+//
+
+type Cgroups struct {
+	cgroupv1 Cgroup
+	cgroupv2 Cgroup
+	cgroup   *Cgroup // pointer to default cgroup version
+	hid      int     // default cgroup controller hiearchy ID
+}
+
+func NewCgroups() (*Cgroups, error) {
+	var err error
+	var cgrp *Cgroup
+	var cgroupv1, cgroupv2 Cgroup
+
+	// discover the default cgroup being used
+	defaultVersion, err := GetCgroupDefaultVersion()
+	if err != nil {
+		return nil, err
```

Comment:
I believe for not to fail in the inexistence of a cgroupfs we should review containers.go as it might depend on its existence (the change would have to be coordinated among the two). I have opened https://github.com/aquasecurity/tracee/issues/2371 to be checked after this PR is merged (it is blocking the new network code for sometime now).